realsense2_camera/src/rs_node_setup.cpp

```diff
@@ -215,7 +221,13 @@ void BaseRealSenseNode::startPublishers(const std::vector<stream_profile>& profi
         rmw_qos_profile_t qos = sensor.getQOS(sip);
         rmw_qos_profile_t info_qos = sensor.getInfoQOS(sip);
 
-        if (profile.is<rs2::video_stream_profile>())
+        // special handling for labeled point cloud stream
+        if (profile.is<rs2::video_stream_profile>() && profile.stream_type() == RS2_STREAM_LABELED_POINT_CLOUD)
+        {
+            _labeled_pointcloud_publisher = _node.create_publisher<sensor_msgs::msg::PointCloud2>("labeled_point_cloud/points",
+                rclcpp::QoS(rclcpp::QoSInitialization::from_rmw(qos),qos));
```

Comment:
Please explain what is QoS?

Reply:
Quality Of Service.

Each publisher should define quality of service, regarding its Durability, Reliability, Depth, History, Queue Size, etc..
Reference: https://docs.ros.org/en/rolling/Concepts/About-Quality-of-Service-Settings.html

Read both parts:
1- https://docs.ros.org/en/rolling/Concepts/About-Quality-of-Service-Settings.html#qos-policies
2- https://docs.ros.org/en/rolling/Concepts/About-Quality-of-Service-Settings.html#qos-profiles